include/uapi/linux/cartesi/cmio.h

```diff
@@ -0,0 +1,40 @@
+/* SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note */
+/*
+ * Cartesi cmio device.
+ * Copyright (C) 2023-2024 Cartesi Machine reference unit
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published by
+ * the Free Software Foundation.
+ */
+
+#ifndef _UAPI_LINUX_CARTESI_CMIO_H
+#define _UAPI_LINUX_CARTESI_CMIO_H
+#include <linux/types.h>
+#include <linux/ioctl.h>
+
+struct cmio_buffer {
+	__u64 data;
```

Comment:
This is supposed to be a pointer, right? Is it common practice to declare it as an integer? Sorry for the ignorance.

Reply:
Yes. Besides this value has to be `mmaped` before used.
I've seen this a ton. IOCTLs `arg` is a integer as well.

Makes the struct layout architecture independent.
Forces you to think about userspace vs kernel address spaces as well. i.e. `copy_to_user((void __user *)arg, ...)`